provider/tencentcloud/tencent_cloud.go

```diff
@@ -0,0 +1,104 @@
+package tencentcloud
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	log "github.com/sirupsen/logrus"
+	"io/ioutil"
+	"sigs.k8s.io/external-dns/endpoint"
+	"sigs.k8s.io/external-dns/plan"
+	"sigs.k8s.io/external-dns/provider"
+	"sigs.k8s.io/external-dns/provider/tencentcloud/cloudapi"
+	"strings"
+)
+
+const (
+	TencentCloudEmptyPrefix = "@"
+	DefaultAPIRate          = 9
+)
+
+func NewTencentCloudProvider(domainFilter endpoint.DomainFilter, zoneIDFilter provider.ZoneIDFilter, configFile string, zoneType string, dryRun bool) (*TencentCloudProvider, error) {
+	cfg := tencentCloudConfig{}
+	if configFile != "" {
+		contents, err := ioutil.ReadFile(configFile)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read Tencent Cloud config file '%s': %v", configFile, err)
```

Comment:
Please use always:
```
fmt.Errorf("....: %w", err)
```
instead of %v

Reply:
thank you, the above all points. I will check it soon.